3_cluster/src/seasonal_metric_cluster.R

```diff
@@ -0,0 +1,275 @@
+    #There will be 2 elements after splitting
+    k <- unlist(strsplit(colnames(cluster_table[,-1]), 
+                         split = '_k'))[seq(2,ncol(cluster_table[,-1])*2,2)] %>%
+      as.numeric()
+    
+    #get all directories to create based on the number of clusters
+    dir_out <- paste0(dir_out, 'cluster', k, '/')
+    
+    fileout <- vector('character', length = sum(k))
+    
+    for (i in 1:length(k)){
+      dir.create(dir_out[i], showWarnings = FALSE)
+      for (cl in 1:k[i]){
+        #metric matrix for gages in cluster
+        metric_mat_c <- filter(metric_mat, 
+                               site_num %in% cluster_table$ID[cluster_table[,i+1] == cl]) %>%
+          select(-site_num)
+        
+        #file index
+        ind_f <- ifelse(test = i > 1, cl + cumsum(k)[i-1], cl)
+        
+        fileout[ind_f] <- paste0(dir_out[i], 
+                                        'SeasonalBarplot_', colnames(cluster_table)[i+1], 
+                                        '_c', cl, '.png')
+        
+        png(filename = fileout[ind_f], width = 5, height = 5, units = 'in', res = 200)
+        barplot(height = colMeans(metric_mat_c), width = 1, 
+                names.arg = season_months, 
+                xlim = c(0,4), ylim = c(0,1),
+                space = 0, main = paste('Metric', metric, ', k = ', k[i], ', Cluster', cl), 
+                xlab = 'Season Months', ylab = 'Seasonal Fraction')
+        #add error bars as 5th - 95th percentiles
+        arrows(x0 = c(0.5,1.5,2.5,3.5), 
+               y0 = as.numeric(apply(X = metric_mat_c, MARGIN = 2, FUN = quantile, 
+                                     probs = 0.05)),
+               x1 = c(0.5,1.5,2.5,3.5), 
+               y1 = as.numeric(apply(X = metric_mat_c, MARGIN = 2, FUN = quantile, 
+                                     probs = 0.95)),
+               angle = 90, length = 0.1, code = 3)
+        dev.off()
+      }
+    }
+  }else{
+    fileout <- paste0(dir_out, 'SeasonalBarplot_', metric, '.png')
+    png(filename = fileout, width = 5, height = 5, units = 'in', res = 200)
+    barplot(height = colMeans(metric_mat[,-1]), width = 1, 
+            names.arg = season_months, 
+            xlim = c(0,4), ylim = c(0,1),
+            space = 0, main = metric, 
+            xlab = 'Season Months', ylab = 'Seasonal Fraction')
+    #add error bars as 5th - 95th percentiles
+    arrows(x0 = c(0.5,1.5,2.5,3.5), 
+           y0 = as.numeric(apply(X = metric_mat[,-1], MARGIN = 2, FUN = quantile, 
+                                 probs = 0.05)),
+           x1 = c(0.5,1.5,2.5,3.5), 
+           y1 = as.numeric(apply(X = metric_mat[,-1], MARGIN = 2, FUN = quantile, 
+                                 probs = 0.95)),
+           angle = 90, length = 0.1, code = 3)
+    dev.off()
+  }
+  
+  return(fileout)
+}
+
+
+#Function to plot the cuts in trees from kmin to kmax clusters
+#clusts is the output from hclust
+plot_cuttree <- function(clusts, kmin, kmax, seq_by, dir_out){
```

Comment:
No, I was debating including it in the diagnostic plot but decided against it. It plots rectangles on the dendrogram around each cluster of gages for a total number of clusters equal to k. I'll leave it included because it can be helpful to visualize the cluster groups after we select how many clusters to use